middle_end/flambda2.0/from_lambda/lambda_to_flambda_primitives.ml

```diff
@@ -349,27 +357,33 @@ let convert_lprim ~backend (prim : L.primitive) (args : Simple.t list)
     let block_access : P.Block_access_kind.t =
       (* Pfield_computed is only used for class access, on blocks of tag 0.
          Obj.field uses Parrayref. *)
-      Block { elt_kind = Value Anything; tag = Tag.zero; size = Unknown; }
+      (* CR mshinwell: For robustness, Pfield_computed should carry the
+         tag, as this condition is not obvious. *)
```

Comment:
If I remember correctly, `Pfield_computed` is just a hack to generate array loads without the float array logic getting in the way. The values read from are created using array primitives, so I would prefer if it was translated to array loads instead of block loads.

Reply:
It looks like these values are actually created in `CamlinternalOO` and have tag `Object_tag`.